### PR TITLE
CASSANDRA-15386 - default to 3 datadirs during in-jvm dtests

### PR DESCRIPTION
**Pull request by krummas**

### CODE REVIEW

File: src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java [hunk 1]
```diff
@@ -268,6 +274,14 @@ public B withInstanceInitializer(BiConsumer<ClassLoader, Integer> instanceInitia
         return (B) this;
     }
 
+    public B withDataDirCount(int datadirCount)
+    {
+        assert datadirCount > 0;
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java:279]: would be nice to leave a message, like `"data dir count requires a positive number but given " + datadirCount`

File: src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java [hunk 2]
```diff
@@ -54,6 +54,7 @@
     private ClassLoader sharedClassLoader = Thread.currentThread().getContextClassLoader();
     private int broadcastPort = 7012;
     private BiConsumer<ClassLoader, Integer> instanceInitializer = (cl, id) -> {};
+    private int datadirCount = 3;
```
Comment (dcapwell) [inline src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java:57]: not sure if we should stay at 1 (match previous logic) or keep this... would make sure tests run with multi dir by default

Comment (krummas) [reply src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java:57]: yeah I want it to be 3 by default to make sure we exercise the range splitting code

File: src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java (hunk 1 above)
Comment (krummas) [reply src/main/java/org/apache/cassandra/distributed/shared/AbstractBuilder.java]: fixed

**Pull request merged**